### PR TITLE
Various fixes

### LINKED ISSUE #38: Build rosetta-helium image from scratch
Currently the rosetta-helium image still uses blockchain-node image as runner image.

For example, in the mainnet Dockerfile: https://github.com/helium/rosetta-helium/blob/main/Dockerfile#L2, it uses `blockchain-node:blockchain-node-ubuntu18-1.1.47` from `quay.io`.

We need to build the rosetta-helium image from scratch for both mainnet and testnet for security reasons.

### LINKED ISSUE #39: Node is always catching up to snapshot height on testnet
I tried the latest version of testnet (blockchain-node-testnet-ubuntu18-1.1.53), but I always hit the following error when I call endpoint `/network/status`
```
{
    "code": 12,
    "message": "Node is not ready",
    "retriable": false,
    "details": {
        "context": "node is catching up to snapshot height"
    }
}
```
I think it hits the error from source code here: https://github.com/helium/rosetta-helium/blob/bump_snap/services/network_service.go#L70 which means the current block height of the node is less than the LBS (I shell into the node container and find that is 1108081, I think it’s the height of the snapshot), I keeping hit this error for about ~3 hours.

@syuan100 thinks the Dockerfile is misconfigured and there shouldn't be a LBS file.

### PR DESCRIPTION
**Pull request by syuan100**
- Build from scratch for node (closes #38)
- Fix Dockerfile for testnet (closes #39)
- Fix timestamp issue for genesis block with no timestamp
- Rework Dockerfile for build-arg
- Add Dockerfile_quick for option of using Helium base images
- Readme updates
- Update ubuntu images to 20.04

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,121 @@
...
+ENV CC=gcc CXX=g++ CFLAGS="-U__sun__" \
+    ERLANG_ROCKSDB_OPTS="-DWITH_BUNDLE_SNAPPY=ON -DWITH_BUNDLE_LZ4=ON" \
+    ERL_COMPILER_OPTIONS="[deterministic]" \
+    PATH="/root/.cargo/bin:/opt/cmake/bin:$PATH" \
+    RUSTFLAGS="-C target-feature=-crt-static"
+
+WORKDIR /usr/src/
 
+# Add our code
+RUN git clone https://github.com/helium/blockchain-node.git
```
Comment (abhay) [inline Dockerfile:89]: This seems like overkill. if a user wants to do a full build, they can pass in a separate container as a base container right?

Comment (syuan100) [reply Dockerfile:89]: Hm I'm not entirely tracking here. From my understanding we need the Dockerfile node to build from source without any dependencies. Wouldn't passing in other containers be considered an outside dependency?

Comment (GeekArthur) [reply Dockerfile:89]: I think @syuan100 is correct, we have to build from source as we can't trust any base container that includes the dependencies.

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,121 @@
 ARG UBUNTU_BUILDER=ubuntu:18.04
```
Comment (GeekArthur) [inline Dockerfile:1]: Please use `Ubuntu 20.04` as base image

**Review by GeekArthur — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (jadeallenx) [reply Dockerfile:1]: ok sounds good

File: Dockerfile (hunk 1 above)
Comment (abhay) [reply Dockerfile:89]: @GeekArthur fair but if you're providing the build image from another source as a build arg, that satisfies your requirements too right? 

My proposal is that you run two builds:

1. blockchain-node (like we do and publish to your local docker repo)
2. rosetta-helium (and you pass in your reference as a UBUNTU_RUNNER). 

That keeps this repo's Dockerfile smaller and doesn't require a full rebuild if folks want to "trust" the helium core team's UBUNTU_RUNNER image.

File: Dockerfile (hunk 2 above)
Comment (abhay) [reply Dockerfile:1]: @GeekArthur can we update this in blockchain-node as well? https://github.com/helium/blockchain-node/blob/master/.buildkite/Dockerfile-ubuntu-erlang#L1

Assuming your preference is ubuntu 20 now, we're happy to not publish ubuntu 18 artifacts.

File: Dockerfile (hunk 1 above)
Comment (syuan100) [reply Dockerfile:89]: I see the value in this approach. However I do also see that in the Rosetta docs it is required to be able to run all components from just one Dockerfile. Maybe we can add a `Dockerfile_quick` to allow for bypassing a from scratch build and utilize the other image builds for speed.

Comment (syuan100) [reply Dockerfile:89]: I think I will also try to rework the main Dockerfile to take an arg for testnet as opposed to having a separate Dockerfile.

Comment (abhay) [reply Dockerfile]: Okay. I'm convinced by the rosetta docs argument!

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,139 @@
...
+ENV CC=gcc CXX=g++ CFLAGS="-U__sun__" \
+    ERLANG_ROCKSDB_OPTS="-DWITH_BUNDLE_SNAPPY=ON -DWITH_BUNDLE_LZ4=ON" \
+    ERL_COMPILER_OPTIONS="[deterministic]" \
+    PATH="/root/.cargo/bin:/opt/cmake/bin:$PATH" \
+    RUSTFLAGS="-C target-feature=-crt-static"
+
+WORKDIR /usr/src/
+
+# Add our code
+RUN git clone https://github.com/syuan100/blockchain-node.git
```
Comment (abhay) [inline Dockerfile:87]: needs update. unless you're waiting on particular merges.

Comment (abhay) [reply Dockerfile:87]: Ah i see it below "fix-historic-balance". Hopefully will get @madninja to release his review block tonight or tomorrow AM.

https://github.com/helium/blockchain-node/pull/85

**Comment by syuan100** [comment]
Oops forgot to fix before pushing. Will add helium repo back after some final testing I'm doing.

File: Dockerfile (hunk 3 above)
Comment (syuan100) [reply Dockerfile:87]: Oops forgot to fix before pushing. Will add helium repo back after some final testing I'm doing.

File: Dockerfile (hunk 1 above)
Comment (GeekArthur) [reply Dockerfile:87]: @abhay Yeah, we have to follow the security rules from Rosetta, but thanks for the proposal.

File: Dockerfile (hunk 2 above)
Comment (GeekArthur) [reply Dockerfile]: @abhay Yep, we can update this.

**Comment by syuan100** [comment]
@abhay one last look, it seems like the node PR was merged so this is ready

**Review by jadeallenx — APPROVED** [review]

**Pull request merged**